### PR TITLE
update MD file and tree structures to reflect YANG model update

### PR DESCRIPTION
**Pull request by QiufangMa**

### CODE REVIEW

File: yang/tree/ical-grp.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    |     +--:(explicit)
+    |     |  +-- period-end?       yang:date-and-time
+    |     +--:(duration)
+    |        +-- duration?         duration
+    +-- bysecond*           uint32
+    +-- byminute*           uint32
+    +-- byhour*             uint32
+    +-- byday* [weekday]
+    |  +-- direction*   int32
+    |  +-- weekday?     schedule:weekday
```
Comment (boucadair) [inline yang/tree/ical-grp.txt:42]:
```suggestion
    |  +-- weekday     schedule:weekday
```

This is a mandatory leaf

File: yang/tree/ical-grp.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    |  +-- time-zone-identifier?   sys:timezone-name
+    +-- frequency?          identityref
+    +-- interval?           uint32
+    +-- (recurrence-bound)?
+    |  +--:(until)
+    |  |  +-- until?        yang:date-and-time
+    |  +--:(count)
+    |     +-- count?        uint32
+    +-- period* [period-start]
+    |  +-- period-start?           yang:date-and-time
```
Comment (boucadair) [inline yang/tree/ical-grp.txt:30]:
```suggestion
    |  +-- period-start            yang:date-and-time
```

This is a list key.

File: yang/tree/rec-tz-dt-grp.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    |  +-- time-zone-identifier?   sys:timezone-name
+    +-- frequency?          identityref
+    +-- interval?           uint32
+    +-- (recurrence-bound)?
+    |  +--:(until)
+    |  |  +-- until?        yang:date-and-time
+    |  +--:(count)
+    |     +-- count?        uint32
+    +-- period* [period-start]
+       +-- period-start?           yang:date-and-time
```
Comment (boucadair) [inline yang/tree/rec-tz-dt-grp.txt:28]:
```suggestion
       +-- period-start            yang:date-and-time
```

This one is mandatory

File: yang/tree/rec-utc-dt-grp.txt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    |  +-- duration?          yang:seconds32
+    +-- frequency?          identityref
+    +-- interval?           uint32
+    +-- (recurrence-bound)?
+    |  +--:(until)
+    |  |  +-- until?        yang:date-and-time
+    |  +--:(count)
+    |     +-- count?        uint32
+    +-- period-timeticks* [period-start]
+       +-- period-start?   yang:timeticks
```
Comment (boucadair) [inline yang/tree/rec-utc-dt-grp.txt:25]:
```suggestion
       +-- period-start    yang:timeticks
```

**Review by boucadair — CHANGES_REQUESTED** [review]
some broken pyang trees (I think this is a bug in pyang)

File: yang/tree/ical-grp.txt (hunk 1 above)
Comment (kwatsen) [reply yang/tree/ical-grp.txt:42]: https://github.com/mbj4668/pyang/issues/902

**Pull request merged**